laws/src/main/scala/org/http4s/laws/discipline/ArbitraryInstances.scala

```diff
@@ -166,27 +166,27 @@ private[http4s] trait ArbitraryInstances extends Ip4sArbitraryInstances {
   implicit val http4sTestingCogenForStatus: Cogen[Status] =
     Cogen[Int].contramap(_.code)
 
-  implicit val http4sTestingArbitraryForQueryParam: Arbitrary[(String, Option[String])] =
+  implicit val http4sTestingArbitraryForQueryParam: Arbitrary[Query.Component] =
     Arbitrary {
       frequency(
         5 -> {
           for {
             k <- getArbitrary[String]
             v <- getArbitrary[Option[String]]
-          } yield (k, v)
+          } yield Query.Component(k, v)
         },
-        2 -> const(("foo" -> Some("bar"))) // Want some repeats
+        2 -> const(Query.Component("foo", "bar")) // Want some repeats
       )
     }
```

Comment:
This only generates decoded components. It should probably generate encoded components too.